src/panels/lovelace/cards/hui-gauge-card.js

```diff
@@ -0,0 +1,182 @@
+          top: calc(var(--base-unit) * 1.2);
+          margin-left: auto;
+          margin-right: auto;
+          transition: all 1s ease-out;
+        }
+        .gauge-data #percent{
+          font-size: calc(var(--base-unit) * 0.55);
+        }
+        .gauge-data #title{
+          padding-top: calc(var(--base-unit) * 0.15);
+          font-size: calc(var(--base-unit) * 0.30);
+        }
+      </style>
+      <ha-card on-click='_handleClick'>
+        <div class='container'>
+          <div class='gauge-a'></div>
+          <div class='gauge-b'></div>
+          <div class='gauge-c' id='gauge'></div>
+          <div class='gauge-data'><div id='percent'></div><div id='title'></div></div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: '_hassChanged'
+      },
+      _config: Object,
+    };
+  }
+
+  ready() {
+    super.ready();
+    if (this._config) this._buildConfig();
+  }
+
+  getCardSize() {
+    return 1;
+  }
+
+  setConfig(config) {
+    this._config = Object.assign({ min: 0, max: 100 }, config);
+    if (!config || !config.entity) throw new Error('Invalid card configuration');
+
+    if (this.$) this._buildConfig();
+    if (this.hass) this._hassChanged(this.hass);
+  }
+
+  _buildConfig() {
+    const entityStateObj = this.hass.states[this._config.entity];
+    const unitOfMeasurement = this._config.unit_of_measurement || entityStateObj.attributes.unit_of_measurement || '';
+    const root = this.shadowRoot;
+
+    if (!this.entityStateObj || entityStateObj.state !== this._entityState.state) {
+      root.getElementById('percent').textContent = `${entityStateObj.state} ${unitOfMeasurement}`;
+      root.getElementById('title').textContent = this._config.title;
+      const turn = this._translateTurn(entityStateObj.state, this._config) / 10;
+      root.getElementById('gauge').style.transform = `rotate(${turn}turn)`;
+      root.getElementById('gauge').style.backgroundColor = this._computeSeverity(entityStateObj.state, this._config.severity);
+      this._entityStateObj = entityStateObj;
+    }
+  }
+
+  _hassChanged(hass) {
+    this.hass = hass;
+    this._buildConfig();
+  }
+
+  _computeSeverity(stateValue, sections) {
+    const numberValue = Number(stateValue);
+    const severityMap = {
+      red: 'var(--label-badge-red)',
+      green: 'var(--label-badge-green)',
+      amber: 'var(--label-badge-yellow)',
+      normal: 'var(--label-badge-blue)',
+    };
+    if (!sections) return severityMap.normal;
+    const sortable = [];
+    Object.keys(sections).forEach((severity) => {
+      sortable.push([severity, sections[severity]]);
+    });
+    sortable.sort((a, b) => a[1] - b[1]);
+
+    if (numberValue >= sortable[0][1] && numberValue < sortable[1][1]) {
+      return severityMap[sortable[0][0]];
+    }
+    if (numberValue >= sortable[1][1] && numberValue < sortable[2][1]) {
```

Comment:
What if user doesn't specify all 3 severity levels? Is that a rule that should be enforced?

Reply:
What's the best way to log an error outside of `setConfig`?  Throwing an Error within `_buildConfig` cause issues because `_hassChanged` calls `_buildConfig` and therefore disables navigation/other activity.

Reply:
Config validation should be done in `setConfig` and errors should be thrown there. If you are going to enforce that all 3 values are supplied, the check should be done any time the card receives a new `config`

Reply:
These are not necessarily config issues because they might change at the next update, so you shouldn't throw, but the card should handle these failures gracefully.

For non-numeric something like: `if (isNaN(numberValue)) return severityMap.normal`

For undefined `stateObj`, your card should have a way to indicate that the entity doesn't exist - similar to the way the entities card renders highlighted error rows if an entity doesn't exist